src/sentry/static/sentry/app/types/index.tsx

```diff
@@ -1028,6 +1028,7 @@ export type SavedQueryState = {
 export type SelectValue<T> = {
   label: string;
   value: T;
+  disabled?: boolean;
```

Comment:
Another approach to this could be to make `OptionSelector` have a type that is `type Option = SelectValue & {disabled?: boolean};`

Reply:
This is the only place `disabled` is used. So the choice here is either to add an optional `disabled` to all `SelectValue` or make `Option` exported and update this PR accordingly.

Reply:
Hrm. Lets go with what you have.